src/lib/optimizer/strategy/join_ordering_rule.cpp

```diff
@@ -69,7 +69,11 @@ std::shared_ptr<AbstractLQPNode> JoinOrderingRule::_perform_join_ordering_recurs
    */
   // TODO(anybody) Increase X once our costing/cardinality estimation is faster/uses internal caching
   auto result_lqp = std::shared_ptr<AbstractLQPNode>{};
-  if (join_graph->vertices.size() < 9) {
+  DebugAssert(join_graph->vertices.size() != 0, "There should be nodes in the join graph.");
+  if (join_graph->vertices.size() == 1) {
```

Comment:
I don't know how this whole mechanism works, but what happens if we have 2 vertexes? Isn't that just a simple join that cannot be reordered? So can't we take the short cut `if vertices.size < 3`?

Reply:
Not necessarily, as JoinOrdering is also responsible for bringing cross joins and their predicates together. Good point though, we should probably add a comment to that regard.